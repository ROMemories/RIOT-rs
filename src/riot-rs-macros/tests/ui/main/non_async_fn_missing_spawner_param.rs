#![no_main]
#![feature(type_alias_impl_trait)]
#![feature(used_with_arg)]

// FAIL: non-async function is expected to take a Spawner as first parameter
#[riot_rs::main]
fn main(foo: Bar) {}

struct Bar;
