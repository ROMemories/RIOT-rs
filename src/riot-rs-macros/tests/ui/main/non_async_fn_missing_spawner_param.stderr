error: custom attribute panicked
 --> tests/ui/main/non_async_fn_missing_spawner_param.rs:6:1
  |
6 | #[riot_rs::main]
  | ^^^^^^^^^^^^^^^^
  |
  = help: message: the function must be async
